File: read_swebok.js
const fs = require('fs');
let pdf = require('pdf-parse');

if (typeof pdf !== 'function' && pdf.default) {
    pdf = pdf.default;
}

const dataBuffer = fs.readFileSync('d:\\repositorio_geral\\app_gestao_contratos\\6 - IEEE_2024_SWEBOK_4.pdf');

pdf(dataBuffer).then(function(data) {
    // Write just the first 50,000 characters to see the index/intro
    fs.writeFileSync('d:\\repositorio_geral\\app_gestao_contratos\\swebok_intro.txt', data.text.substring(0, 50000));
    // Write whole file for reference
    fs.writeFileSync('d:\\repositorio_geral\\app_gestao_contratos\\swebok_full.txt', data.text);
    console.log('✅ SWEBOK PDF text extracted!');
}).catch(err => {
    console.error('Erro na extração do SWEBOK:', err);
});

File: read_swebok.py
import sys

try:
    import PyPDF2
    print("PyPDF2 is installed!")
    reader = PyPDF2.PdfReader('6 - IEEE_2024_SWEBOK_4.pdf')
    text = ""
    for i in range(min(50, len(reader.pages))):
        text += reader.pages[i].extract_text() + "\n"
    with open('swebok_intro.txt', 'w', encoding='utf-8') as f:
        f.write(text)
    print("✅ SWEBOK PDF text extracted (PyPDF2)!")
except Exception as e:
    print(f"PyPDF2 error: {e}")
    try:
        import pdfplumber
        print("pdfplumber is installed!")
        with pdfplumber.open('6 - IEEE_2024_SWEBOK_4.pdf') as pdf:
            text = ""
            for i in range(min(50, len(pdf.pages))):
                text += pdf.pages[i].extract_text() + "\n"
        with open('swebok_intro.txt', 'w', encoding='utf-8') as f:
            f.write(text)
        print("✅ SWEBOK PDF text extracted (pdfplumber)!")
    except Exception as e2:
        print(f"pdfplumber error: {e2}")
